src/schema/v2/artwork/index.ts

```diff
@@ -154,6 +154,20 @@ export const ArtworkType = new GraphQLObjectType<any, ResolverContext>({
       },
       consignmentSubmission: {
         type: ArtworkConsignmentSubmissionType,
+        resolve: async ({ id }, _options, { submissionsLoader }) => {
+          if (!submissionsLoader) {
+            return
+          }
+
+          const submissions = await submissionsLoader({ size: 1000 })
+          const filteredSubmissions = submissions.filter(
+            (submission) => submission.state !== "draft"
+          )
+
+          return filteredSubmissions.find((submission) => {
+            return submission.my_collection_artwork_id === id
+          })
+        },
```

Comment:
Have you checked how many requests to Convection we will fire when we load many artworks? If it just fires one request, this solution is pretty nice 👍

Reply:
I've checked local metaphysics logs and I see only 1 requests to convection. If you know a better way how to check if the code triggers >1 request - please let me know